builder/build.go

```diff
@@ -152,34 +161,41 @@ type dockerBuild struct {
 	BuildLabelMap    map[string]string
 }
 
-// createBuildTemplate creates temporary build folder to perform a Docker build with language template
-func createBuildTemplate(functionName string, handler string, language string, useFunction bool) (string, error) {
+const defaultHandlerFolder = "function"
+
+// createBuildContext creates temporary build folder to perform a Docker build with language template
+func createBuildContext(functionName string, handler string, language string, useFunction bool, handlerFolder string) (string, error) {
 	tempPath := fmt.Sprintf("./build/%s/", functionName)
 	fmt.Printf("Clearing temporary build folder: %s\n", tempPath)
 
 	clearErr := os.RemoveAll(tempPath)
 	if clearErr != nil {
-		fmt.Printf("Error clearing temporary build folder %s\n", tempPath)
+		fmt.Printf("Error clearing temporary build folder: %s\n", tempPath)
 		return tempPath, clearErr
 	}
 
 	functionPath := tempPath
+
 	if useFunction {
-		functionPath += "/function"
+		if handlerFolder == "" {
+			functionPath = path.Join(functionPath, defaultHandlerFolder)
+		} else {
+			functionPath = path.Join(functionPath, handlerFolder)
```

Comment:
not sure do we want to check for prepended `/` for `handlerFolder`?

Reply:
It should add it through path.Join